doc/specs/stdlib_str2num.md

```diff
@@ -0,0 +1,105 @@
+---
+title: str2num
+---
+
+# The `stdlib_str2num` module
+
+This module proposes a function-style interface for string-to-number conversion. It also profits from Fortran's interfaces to implement precision-dependant algorithms to maximize runtime efficiency.
+
+[TOC]
+
+## `to_num` - conversion of strings to numbers
+
+### Status
+
+Experimental
+
+### Description
+
+Convert a string or an array of strings to numerical types.
+
+### Syntax
+
+`number = [[stdlib_str2num(module):to_num(interface)]](string, mold)`
+
+### Arguments
+
+`string`: argument has `intent(in)` and is of type `character(*)`.
+
+`mold`: argument has `intent(in)` and is of numerical type. currently: `integer`, `real32`, `real64` or `real128`. **Note**: the mold argument is included to help compilers chose the correct implementation at compile-time. Currently, compilers are not able to disambiguate functions with respect to the left-hand-side of an assignment.
```

Comment:
All interpretation of expressions in Fortran and most other programming languages is context-free.  This is not a temporary inability of compilers to perform an analysis, and there are far more possible contexts than being the entire right-hand side of an expression.

Reply:
Actually, because all types of `integer` and `real` can be now supported (through fypp preprocessing) the following should be sufficient (similarly as in other specs):
```suggestion
`mold`: argument has `intent(in)` and is of numerical type (that is of `integer` or of `real32`). **Note**: the mold argument is included to help compilers chose the correct implementation at compile-time. Currently, compilers are not able to disambiguate functions with respect to the left-hand-side of an assignment.
```

Reply:
Agreed, I felt some comment was needed in order to justify the requirement of a mold argument. What would you suggest as a modification to this comment?

Reply:
Additionally, with the latest changes, kinds are defined in `stdlib_kinds` and not in `iso_fortran_env`, where `stdlib_kinds` defines its own `sp`, `dp` and `qp`.

Reply:
"The type of the `mold` argument defines the type of the result."